gr-utils/python/blocktool/core/parseheader.py

```diff
@@ -0,0 +1,281 @@
+        """ Override the Blocktool validate function """
+        BlockTool._validate(self)
+        if self.info['modname'].startswith(Constants.GR):
+            _module = self.info['modname'].split('-')[-1]
+        else:
+            _module = self.info['modname']
+            self.info['modname'] = Constants.GR+_module
+        if _module not in self.module_types:
+            raise BlockToolException('Module must be one of {}.'.format(
+                ', '.join(self.module_types)))
+        if not self.info['filename'].endswith('.h'):
+            raise BlockToolException(
+                'Invalid header file')
+        _target_dir = os.path.join(self.target_dir,
+                                   self.info['modname'],
+                                   'include',
+                                   'gnuradio',
+                                   self.info['modname'].split('-')[-1])
+        if _target_dir != self.info['target_dir']:
+            raise BlockToolException('Invalid GNU Radio module')
+        try:
+            open(self.info['impl_file'], 'r')
+        except OSError:
+            raise OSError
+
+    def get_header_info(self):
+        """
+        PyGCCXML header code parser
+        magic happens here!
+        # : returns the parsed header data in python dict
+        # : return dict keys: namespace, class, io_signature, make,
+        #                properties, methods
+        # : Can be used as an CLI command or an extenal API
+        """
+        gr = self.info['modname'].split('-')[0]
+        module = self.info['modname'].split('-')[-1]
+        generator_path, generator_name = utils.find_xml_generator()
+        xml_generator_config = parser.xml_generator_configuration_t(
+            xml_generator_path=generator_path,
+            xml_generator=generator_name,
+            compiler='gcc')
+        decls = parser.parse(
+            [self.info['target_file']], xml_generator_config)
+        global_namespace = declarations.get_global_namespace(decls)
+
+        # namespace
+        try:
+            self.parsed_data['namespace'] = []
+            ns = global_namespace.namespace(gr)
+            if ns is None:
+                raise BlockToolException
+            main_namespace = ns.namespace(module)
+            if main_namespace is None:
+                raise BlockToolException('namespace cannot be none')
+            self.parsed_data['namespace'] = [gr, module]
+            if main_namespace.declarations:
+                for _namespace in main_namespace.declarations:
+                    if isinstance(_namespace, declarations.namespace_t):
+                        if Constants.KERNEL not in str(_namespace):
+                            main_namespace = _namespace
+                            self.parsed_data['namespace'].append(
+                                str(_namespace).split('::')[-1].split(' ')[0])
+        except BlockToolException as exception:
+            raise BlockToolException(
```

Comment:
This looks weird to me, shouldn't this be another type of exception

Reply:
Changed to RuntimeError